src/helpers.ts

```diff
@@ -0,0 +1,23 @@
+import { assert } from 'superstruct';
+
+import { EthAccountType } from './eth';
+import type { InternalAccount } from './internal';
+import { InternalAccountStruct } from './internal';
+
+/**
+ * Checks if the given internal account is of the EVM account type.
+ * @param account - The internal account to check.
+ * @returns Returns true if the internal account has an EVM account type, false otherwise.
+ */
+export function isEVMAccount(account: InternalAccount): boolean {
+  // using try catch here to avoid throwing an error because the internal account's define would throw the error.
```

Comment:
Maybe we could use `superstruct.is` in this case?

Reply:
The `define` would throw the error, so it doesn't really matter if we use `is`

Reply:
Hm I think a wrapper is beneficial to handle some edge cases

Reply:
But I don't have a strong opinion about it

Reply:
`define` for `InternalAccount` shouldn't throw, it either returns a `StructError` or a `boolean`

This is to make sure it's compliant for `superstruct.is` (at least, from superstruct doc, this is how it should be implemented)

Reply:
Updated the helper to take the account type instead of the account as an argument